src/algorithms/telemetry_decoder/gnuradio_blocks/sbas_l1_telemetry_decoder_gs.cc

```diff
@@ -41,20 +42,23 @@
 
 sbas_l1_telemetry_decoder_gs_sptr sbas_l1_make_telemetry_decoder_gs(
     const Gnss_Satellite &satellite,
-    bool dump)
+    const Tlm_Conf &conf)
 {
-    return sbas_l1_telemetry_decoder_gs_sptr(new sbas_l1_telemetry_decoder_gs(satellite, dump));
+    return sbas_l1_telemetry_decoder_gs_sptr(new sbas_l1_telemetry_decoder_gs(satellite, conf));
 }
 
 
 sbas_l1_telemetry_decoder_gs::sbas_l1_telemetry_decoder_gs(
     const Gnss_Satellite &satellite,
-    bool dump) : gr::block("sbas_l1_telemetry_decoder_gs",
-                     gr::io_signature::make(1, 1, sizeof(Gnss_Synchro)),
-                     gr::io_signature::make(1, 1, sizeof(Gnss_Synchro))),
-                 d_dump(dump),
-                 d_channel(0),
-                 d_block_size(D_SAMPLES_PER_SYMBOL * D_SYMBOLS_PER_BIT * D_BLOCK_SIZE_IN_BITS)
+    const Tlm_Conf &conf) : gr::block("sbas_l1_telemetry_decoder_gs",
+                                gr::io_signature::make(1, 1, sizeof(Gnss_Synchro)),
+                                gr::io_signature::make(1, 1, sizeof(Gnss_Synchro))),
+                            d_dump(true),
+                            d_dump_mat(true),
+                            d_remove_dat(true),
+                            d_dump_filename("tele"),
+                            d_channel(0),
```

Comment:
I don't think these should be hard-coded. You created a config object; hopefully it encompasses these values. Use the config here.

Note that if (in particular) `d_channel` is hard-coded to 0, why bother keeping the state in this object?